examples/rancher-gen/nginx.tmpl

```diff
@@ -0,0 +1,183 @@
+{{$local := host}}
+#This Host:
+#  UUID: {{$local.UUID}}
+#  Name: {{$local.Name}}
+#  Address: {{$local.Address}}
+#  Hostname: {{$local.Hostname}}
+#  Labels: {{$local.Labels}}
+
+# Access Log Formatting
+log_format vhost '$host $remote_addr - $remote_user [$time_local] '
+                 '"$request" $status $body_bytes_sent '
+                 '"$http_referer" "$http_user_agent"';
+
+
+# If we receive X-Forwarded-Proto, pass it through; otherwise, pass along the
+# scheme used to connect to this server
+map $http_x_forwarded_proto $proxy_x_forwarded_proto {
+  default $http_x_forwarded_proto;
+  ''      $scheme;
+}
+
+# If we receive X-Forwarded-Port, pass it through; otherwise, pass along the
+# server port the client connected to
+map $http_x_forwarded_port $proxy_x_forwarded_port {
+  default $http_x_forwarded_port;
+  ''      $server_port;
+}
+
+# If we receive Upgrade, set Connection to "upgrade"; otherwise, delete any
+# Connection header that may have been passed to this server
+map $http_upgrade $proxy_connection {
+  default upgrade;
+  '' close;
+}
+
+gzip_types text/plain text/css application/javascript application/json application/x-javascript text/xml application/xml application/xml+rss text/javascript;
```

Comment:
Two proposals:

1. Move `gzip_types` to `server{}` directives ([docs](http://nginx.org/en/docs/http/ngx_http_gzip_module.html)) ([example](https://github.com/jwilder/docker-gen/blob/master/templates/nginx.tmpl#L46))
1. Implement [vhost overriding](https://github.com/Munsio/docker-letsencrypt-nginx-proxy-companion/blob/master/app/nginx.tmpl#L186-L190)